### DISCUSSION TITLE
How to train on multiple GPUs?

### DISCUSSION THREAD

**Discussion by rbunn80110** [KeremTurgutlu/self_supervised#51] [Q&A] [answered]
I am working with the barlow twins model, but can't seem to figure out how to train on 4 GPUs.  Anyone know how?

**Comment by KeremTurgutlu** [comment, marked as answer]
You can refer to regular fastai docs/forums/discord channel for distributed training in general. [Here](https://github.com/fastai/fastai/blob/ab154927696338741e59e0ffc4774777c4a9781c/nbs/examples/train_imagenette.py#L83) is an example from fastai repo.

**Reply by rbunn80110** [reply]
I just found the answer.  I read through the docs, I was just over complicating things.

BTW, this is really great work that has saved me a ton of time.  Thank you so much!

python -m fastai.launch scriptname.py 
works for me.

**Reply by KeremTurgutlu** [reply]
Yes that launch script is needed. I am very glad it's helpful :)
